internal/engine/experiment/quicping/quicping.go

```diff
@@ -0,0 +1,479 @@
+// https://www.rfc-editor.org/rfc/rfc9000.html#name-version-negotiation-packet
+func (m *Measurer) DissectVersionNegotiation(i []byte) ([]uint32, ConnectionID, error) {
+	firstByte := uint8(i[0])
+	mask := 0b10000000
+	mask &= int(firstByte)
+	if mask == 0 {
+		return nil, nil, &errUnexpectedResponse{msg: "not a long header packet"}
+	}
+
+	versionBytes := i[1:5]
+	v := binary.BigEndian.Uint32(versionBytes)
+	if v != 0 {
+		return nil, nil, &errUnexpectedResponse{msg: "unexpected Version Negotiation format"}
+	}
+
+	dstLength := i[5]
+	offset := 6 + uint8(dstLength)
+	dst := i[6:offset]
+
+	srcLength := i[offset]
+	offset = offset + 1 + srcLength
+
+	n := uint8(len(i))
+	var supportedVersions []uint32
+	for offset < n {
+		supportedVersions = append(supportedVersions, binary.BigEndian.Uint32(i[offset:offset+4]))
+		offset += 4
+	}
+	return supportedVersions, dst, nil
+}
+
+// NewExperimentMeasurer creates a new ExperimentMeasurer.
+func NewExperimentMeasurer(config Config) model.ExperimentMeasurer {
+	return &Measurer{config: config}
+}
+
+// SummaryKeys contains summary keys for this experiment.
+//
+// Note that this structure is part of the ABI contract with probe-cli
+// therefore we should be careful when changing it.
+type SummaryKeys struct {
+	IsAnomaly bool `json:"-"`
+}
+
+// GetSummaryKeys implements model.ExperimentMeasurer.GetSummaryKeys.
+func (m *Measurer) GetSummaryKeys(measurement *model.Measurement) (interface{}, error) {
+	return SummaryKeys{IsAnomaly: false}, nil
+}
+
+// buildHeader creates the unprotected QUIC header.
+// https://www.rfc-editor.org/rfc/rfc9000.html#name-initial-packet
+func buildHeader(destConnID, srcConnID ConnectionID, payloadLen int) []byte {
+	hdr := []byte{0xc3} // long header type, fixed
+
+	version := make([]byte, 4)
+	binary.BigEndian.PutUint32(version, uint32(0xbabababa))
+	hdr = append(hdr, version...) // version
+
+	lendID := uint8(len(destConnID))
+	hdr = append(hdr, lendID)        // destination connection ID length
+	hdr = append(hdr, destConnID...) // destination connection ID
+
+	lensID := uint8(len(srcConnID))
+	hdr = append(hdr, lensID)       // source connection ID length
+	hdr = append(hdr, srcConnID...) // source connection ID
+
+	hdr = append(hdr, 0x0) // token length
+
+	remainder := 4 + payloadLen
+	remainder_mask := 0b100000000000000
+	remainder_mask |= remainder
+	remainder_b := make([]byte, 2)
+	binary.BigEndian.PutUint16(remainder_b, uint16(remainder_mask))
+	hdr = append(hdr, remainder_b...) // remainder length: packet number + encrypted payload
+
+	pn := make([]byte, 4)
+	binary.BigEndian.PutUint32(pn, uint32(2))
+	hdr = append(hdr, pn...) // packet number
+
+	return hdr
+}
+
+// buildPacket constructs an Initial QUIC packet
+// and applies Initial protection.
+// https://www.rfc-editor.org/rfc/rfc9001.html#name-client-initial
+func buildPacket() ([]byte, ConnectionID, ConnectionID) {
```

Comment:
Please, move this function to a separate file. Logically, this is a basic building block for the main implementation of the experiment, so I'd like it to be in a separate file. The way I see it, our stack here is like:

```
  experiment (quicping.go)
  --------
  separate file (maybe quic.go?)
  --------
  crypto (crypto.go)

```